File: backend/utils/response.py
from flask import jsonify

def success_response(message, data=None, status_code=200):
    """Generates standard success JSON payload."""
    response = {
        "success": True,
        "message": message
    }
    if data is not None:
        response["data"] = data
        
    return jsonify(response), status_code

def error_response(message, details=None, status_code=400):
    """Generates standard failure/error JSON payload."""
    response = {
        "success": False,
        "message": message
    }
    if details is not None:
        response["details"] = details
        
    return jsonify(response), status_code

File: backend/src/utils/response.ts
import { Response } from 'express';

export interface ApiResponse<T = any> {
  success: boolean;
  message: string;
  data?: T;
  meta?: any;
  error?: any;
}

export const sendSuccess = <T>(
  res: Response,
  message: string,
  data?: T,
  statusCode: number = 200,
  meta?: any
): Response => {
  return res.status(statusCode).json({
    success: true,
    message,
    data,
    meta,
  });
};

export const sendError = (
  res: Response,
  message: string,
  statusCode: number = 500,
  error?: any
): Response => {
  return res.status(statusCode).json({
    success: false,
    message,
    error: error || undefined,
  });
};
